tests/MongoDB.Driver.Tests/Specifications/unified-test-format/EntityMap.cs

```diff
@@ -0,0 +1,478 @@
+                }
+            }
+        }
+
+        public IGridFSBucket GetBucket(string bucketId)
+        {
+            return _buckets[bucketId];
+        }
+
+        public IEnumerator<ChangeStreamDocument<BsonDocument>> GetChangeStream(string changeStreamId)
+        {
+            return _changeStreams[changeStreamId];
+        }
+
+        public IMongoClient GetClient(string clientId)
+        {
+            return _clients[clientId];
+        }
+
+        public IMongoCollection<BsonDocument> GetCollection(string collectionId)
+        {
+            return _collections[collectionId];
+        }
+
+        public IMongoDatabase GetDatabase(string databaseId)
+        {
+            return _databases[databaseId];
+        }
+
+        public EventCapturer GetEventCapturer(string clientId)
+        {
+            return _clientEventCapturers[clientId];
+        }
+
+        public BsonValue GetResult(string resultId)
+        {
+            return _results[resultId];
+        }
+
+        public IClientSessionHandle GetSession(string sessionId)
+        {
+            return _sessions[sessionId];
+        }
+
+        public BsonDocument GetSessionId(string sessionId)
+        {
+            return _sessionIds[sessionId];
+        }
+
+        public bool HasBucket(string bucketId)
+        {
+            return _buckets.ContainsKey(bucketId);
+        }
+
+        public bool HasChangeStream(string changeStreamId)
+        {
+            return _changeStreams.ContainsKey(changeStreamId);
+        }
+
+        public bool HasClient(string clientId)
+        {
+            return _clients.ContainsKey(clientId);
+        }
+
+        public bool HasCollection(string collectionId)
+        {
+            return _collections.ContainsKey(collectionId);
+        }
+
+        public bool HasDatabase(string databaseId)
+        {
+            return _databases.ContainsKey(databaseId);
+        }
+
+        public bool HasSession(string sessionId)
+        {
+            return _sessions.ContainsKey(sessionId);
+        }
+
+        // private methods
+        private IGridFSBucket CreateBucket(BsonDocument entity)
+        {
+            IMongoDatabase database = null;
+
+            foreach (var element in entity)
+            {
+                switch (element.Name)
+                {
+                    case "id":
+                        // handled on higher level
+                        break;
+                    case "database":
+                        var databaseId = element.Value.AsString;
+                        database = _databases[databaseId];
+                        break;
+                    default:
+                        throw new FormatException($"Unrecognized bucket entity field: '{element.Name}'");
+                }
+            }
+
+            return new GridFSBucket(database);
+        }
+
+        private (DisposableMongoClient, EventCapturer) CreateClient(BsonDocument entity)
```

Comment:
Consider:

    private void CreateClient(BsonDocument entity, out DisposableMongoClient client, out EventCapturer eventCapturer)

to avoid annoying `Item1` and `Item2` in tuple result.

Having **two** out parameters clearly indicates that calling this function has **two** results **and** documents what they are.